sw/device/lib/testing/test_coverage.c

```diff
@@ -0,0 +1,79 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/uart.h"
+#include "sw/vendor/llvm_clang_rt_profile/compiler-rt/lib/profile/InstrProfiling.h"
+#include <stdint.h>
+
+/**
+ * When the linker finds a definition of this symbol, it knows to skip loading
+ * the object which contains the profiling runtime’s static initializer.
+ */
+int __llvm_profile_runtime;
+
+/**
+ * Pointer to the LLVM profiling runtime initialization function in
+ * `.init_array`, defined in `flash_link.ld`.
+ */
+extern void (*_init_array_start)();
+
+/**
+ * Computes the CRC32 of a buffer as expected by Python's `zlib.crc32()`. The
+ * implementation below is basically a simplified, i.e. byte-by-byte and without
+ * a lookup table, version of zlib's crc32. See
+ * https://github.com/madler/zlib/blob/2fa463bacfff79181df1a5270fb67cc679a53e71/crc32.c,
+ * lines 111-112 and 276-279.
+ */
+static uint32_t crc32(char *buf, size_t len) {
+  // CRC32 polynomial.
+  static const uint32_t kPoly = 0xEDB88320;
+  // Since we use a contiguous buffer, we don't need to call this function
+  // multiple times. That's why `crc` is not a parameter and is initialized to
+  // zero.
+  uint32_t crc = 0;
+  crc = ~crc;
+  for (; len > 0; --len, ++buf) {
+    crc = crc ^ *buf;
+    for (uint8_t i = 0; i < 8; ++i) {
+      if (crc & 1) {
+        crc = kPoly ^ (crc >> 1);
+      } else {
+        crc = crc >> 1;
+      }
+    }
+  }
+  return ~crc;
+}
+
+/**
+ * Sends the given buffer as a hex string over UART.
+ */
+static void send_buffer(char *buf, size_t len) {
+  for (uint32_t i = 0; i < len; ++i) {
+    base_printf("%2X", buf[i]);
+  }
+}
+
+void test_coverage_init_buffer() { (*_init_array_start)(); }
```

Comment:
We should probably do this correctly, with a for loop between `_init_array_start` and `_init_array_end`, calling each pointer.

Reply:
Thanks for catching this. I remember we talked about this and whether we should do this in `flash_crt.S` instead of a function. I _guess_ it is OK to do this here since we don't have anything else in `init_array` at the moment but let me know if you want me to move this to `flash_crt.S`. Done.

Reply:
I actually ended up moving this to `flash_crt.S`. Please let me know what you think.